### PR TITLE
Improve loop for getAllQueries from FTL's memory

### PR DESCRIPTION
**Pull request by yubiuser**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**
This PR reduces the memory needed by `api.php?getAllQueries` to process every requested record.
It's basically a copy of https://github.com/pi-hole/AdminLTE/pull/2114 but for the current in-memory data served by FTL.

Usually we have seen the out-of-memory error when requesting large amounts of data from the long-term database, but currently there is an issue on discourse where the in-memory data is also too huge to process `getAllQueries`

https://discourse.pi-hole.net/t/all-queries-are-not-working/54711

**How does this PR accomplish the above?:**
The old code used to generate and return the request adds all records into a single array. This array grows for each record (and there are many many records).
Only at the end (and if memory holds up), it transforms the entire array into JSON and sends all at once as a string.

The new code outputs each record as a string immediately.
This way, there is no array and the memory for each iteration is always the same.
The web server is responsible for sending the "text stream", nothing else needs to change.

### CODE REVIEW

**Comment by yubiuser** [comment]
Has been confirmed working in the linked discourse topic.

**Review by DL6ER — COMMENTED** [review]
There are merge conflicts.

I think it's worth noting that this removed a feature of the API that allows combining multiple requests in the same API call. I don't really know if it is every used for this endpoint, but we use it elsewhere. Example: Requests like `api.php?summary&getAllQueries` change behavior in this PR (`summary` is now ignored)

**Comment by yubiuser** [comment]
Merge conflicts resolved.

**DL6ER** made a good point. I searched for `api.php?` within the source and could not find any occurrence with `getAllQueries` and something else. But user scripts might use it. Not sure how to proceed. Is this a show stopper?
_

My last commit (moving the `exit`) made me think if the `exit` in [this PR](https://github.com/pi-hole/AdminLTE/pull/2114/files) is also at the wrong spot? @rdwebdesign

**Review by rdwebdesign — DISMISSED** [review]
Looks fine.

File: api_FTL.php [hunk 1]
```diff
@@ -306,19 +306,35 @@
...
-          $tmp[2] = utf8_encode(str_replace("~"," ",$tmp[2]));
-          // UTF-8 encode client host name
-          $tmp[3] = utf8_encode($tmp[3]);
-          array_push($allQueries,$tmp);
+      // Start the JSON string
+      echo '{"data":[';
+      $first = true;
+
+      #foreach ($return as $line)
+      foreach($return as $line) {
```
Comment (rdwebdesign) [inline api_FTL.php]: Why did you add this comment?

File: api_FTL.php [hunk 2]
```diff
@@ -306,19 +306,35 @@
...
-          // UTF-8 encode domain
-          $tmp[2] = utf8_encode(str_replace("~"," ",$tmp[2]));
-          // UTF-8 encode client host name
-          $tmp[3] = utf8_encode($tmp[3]);
-          array_push($allQueries,$tmp);
+      // Start the JSON string
+      echo '{"data":[';
+      $first = true;
+
+      #foreach ($return as $line)
```
Comment (yubiuser) [inline api_FTL.php]:
```suggestion

```

**Review by rdwebdesign — APPROVED** [review]
